src/xbot/common/properties/Property.java

```diff
@@ -22,17 +22,42 @@
 
     ITableProxy permanentStore;
     ITableProxy randomAccessStore;
-
+    
+    /** 
+     * New enum to determine property persistence
+     */
+    public enum PropertyPersistenceType{
```

Comment:
Yeah,I agree with Alex - factory methods would be pretty readable and autocomplete nicely.